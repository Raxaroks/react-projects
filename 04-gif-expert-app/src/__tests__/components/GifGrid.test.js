import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente: <GifGrid />', () => {

    const category = 'Nekane';
    
    test('Debe de renderizarse y mostrarse correctamente.', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const componentWrapper = shallow(<GifGrid category={category} />);
        expect(componentWrapper).toMatchSnapshot();
    })
    
    test('Debe de mostrar los ítems al cargar las imágenes.', () => {
    
        const gifs = [{
            id: 'ABC',
            url: 'https://somepage.io/anything',
            title: 'Any title'
        },{
            id: '123',
            url: 'https://somepage.io/anything',
            title: 'Any title'
        }
        ]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const componentWrapper = shallow(<GifGrid category={category} />);

        // expect( componentWrapper ).toMatchSnapshot();
        expect( componentWrapper.find('p').exists() ).toBe(false);
        expect( componentWrapper.find('GifGridItem').length ).toBe(gifs.length);
    })
    
})
