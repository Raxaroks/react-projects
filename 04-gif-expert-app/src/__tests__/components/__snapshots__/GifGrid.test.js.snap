// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente: <GifGrid /> Debe de renderizarse y mostrarse correctamente. 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Nekane
  </h3>
  <p>
    Loading ...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
